### PR TITLE
Fix adding item to cart without sufficient stock available

### PR DESCRIPTION
**Pull request by nikunjsdev**
Fix adding item to cart without sufficient stock available

### CODE REVIEW

**Comment by zshanhui** [comment]
I don't see this on the `collections/cabinets/Q29sbGVjdGlvbjo0/dark-elegance/cabinet-parts` page. Did you implement?

**Comment by nikunjsdev** [comment]
> I don't see this on the `collections/cabinets/Q29sbGVjdGlvbjo0/dark-elegance/cabinet-parts` page. Did you implement?

I have implemented this in `collections/cabinets/Q29sbGVjdGlvbjo0/dark-elegance/cabinet-parts` page and pushed the code.

**Comment by zshanhui** [comment]
@nikunjsdev Can you add screenshots of the relevant screens?

File: src/components/Collection/CollectionBlock.tsx [hunk 1]
```diff
@@ -28,7 +28,9 @@ export const CollectionBlock = props => {
           onClick={handleWishlist}
         />
         <Link to={`/collections/cabinets/${collect.id}/${collect.slug}`}>
```
Comment (zshanhui) [inline src/components/Collection/CollectionBlock.tsx:30]: Change `collect` to `collection`

Either use 1 letter: `c` for short lambda functions like `collections.map(c => c.namae)` or use the full name to keep naming consistent.

File: src/components/OverlayManager/Cart/ProductList.tsx [hunk 2]
```diff
@@ -2,54 +2,93 @@ import * as React from "react";
...
-              </p>
-              <QauntSelect
-                id={line.id}
-                quantity={line.quantity}
-                add={add}
-                subtract={subtract}
+}> = ({ lines, remove, add, subtract }) => {
+  const [stockQuantity, setStockQuantity] = React.useState({
+    id: null,
+    stock: 0,
```
Comment (zshanhui) [inline src/components/OverlayManager/Cart/ProductList.tsx:21]: The stock comes from CMS backend, I don't see any reason it should be set to `0` or need an `id`. I think you might be over complicating things here. Please ask me if you have any doubts.

File: src/components/OverlayManager/Cart/ProductList.tsx [hunk 3]
```diff
@@ -2,54 +2,93 @@ import * as React from "react";
...
+  });
+  return (
+    <div>
+      <div className="cart__header">
+        <span>Colored Parrot Cushion</span>
```
Comment (zshanhui) [inline src/components/OverlayManager/Cart/ProductList.tsx:26]: Why is this hardcoded?

File: src/components/OverlayManager/Cart/ProductList.tsx [hunk 4]
```diff
@@ -2,54 +2,93 @@ import * as React from "react";
...
+          const cart =
+            JSON.parse(localStorage.getItem(LocalStorageKeys.Cart)) || [];
+          const cartItem = cart.find(item => item.variantId === line.id);
+          const cartQuantity = (cartItem && cartItem.quantity) || 0;
+          if (stockQuantity.id !== line.id) {
+            setStockQuantity({
+              id: line.id,
+              stock: line.stockQuantity - cartQuantity,
+            });
+          }
```
Comment (zshanhui) [inline src/components/OverlayManager/Cart/ProductList.tsx:40]: I'm not sure what you're doing here, but it doesn't look right. Overcomplicated.

You don't need `localStorage` for this.

You just need to compare the item quantity with the backends' stock number. It's really simple.

**Review by zshanhui — CHANGES_REQUESTED** [review]

**Comment by zshanhui** [comment]
@nikunjsdev Hey, I've reviewed this PR and I think you're overcomplicating this task and going in the wrong direction. Let's put this on hold first, and I will assign you some more UI related tasks as it looks like you handle those better than Javascript heavy logic.

File: src/components/OverlayManager/Cart/ProductList.tsx [hunk 5]
```diff
@@ -2,54 +2,93 @@ import * as React from "react";
 import { TaxedMoney } from "@components/containers";
 import { Thumbnail } from "@components/molecules";
 import { LineI } from "../../CartTable/ProductRow";
+import ReactSVG from "react-svg";
+import editIcon from "../../../images/baseline-edit-24px.svg";
+import cancelIcon from "../../../images/cancel-24px.svg";
+
+enum LocalStorageKeys {
+  Cart = "cart",
+}
```
Comment (zshanhui) [inline src/components/OverlayManager/Cart/ProductList.tsx:11]: No need for `localStorage` here.

**Review by zshanhui — CHANGES_REQUESTED** [review]

**Pull request closed without merging**